blockchain-agent/src/main/java/com/sphereon/alfresco/blockchain/agent/tasks/scheduled/SignNewDocuments.java

```diff
@@ -0,0 +1,50 @@
+package com.sphereon.alfresco.blockchain.agent.tasks.scheduled;
+
+import com.sphereon.alfresco.blockchain.agent.tasks.AlfrescoRepository;
+import com.sphereon.alfresco.blockchain.agent.model.AlfrescoBlockchainRegistrationState;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import static com.sphereon.alfresco.blockchain.agent.model.AlfrescoBlockchainRegistrationState.PENDING_REGISTRATION;
+
+@Component
+public class SignNewDocuments {
+    private static final int EXECUTION_RATE = 30000;
+
+    private static final org.slf4j.Logger logger = LoggerFactory.getLogger(SignNewDocuments.class);
+
+    private final AlfrescoRepository alfrescoRepository;
+    private final SignNewDocumentsTask signNewDocumentsTask;
+
+    public SignNewDocuments(final AlfrescoRepository alfrescoRepository,
+                            final SignNewDocumentsTask signNewDocumentsTask) {
+        this.alfrescoRepository = alfrescoRepository;
+        this.signNewDocumentsTask = signNewDocumentsTask;
+    }
+
+    @Scheduled(fixedRate = EXECUTION_RATE)
+    public synchronized void execute() {
+        try {
+            logger.info("Searching for documents with registration state " + PENDING_REGISTRATION);
+            this.alfrescoRepository.selectEntries(PENDING_REGISTRATION)
+                    .forEach(rowEntry -> {
+                        final var entry = rowEntry.getEntry();
+                        try {
+                            logger.info("Found document " + entry.getName() + " / " + entry.getId());
+                            var contentHash = this.alfrescoRepository.hashEntry(entry.getId());
+                            this.signNewDocumentsTask.registerEntry(contentHash);
+                            logger.info("Updating state to pending for document " + entry.getName() + " / " + entry.getId());
+                            final var registrationState = AlfrescoBlockchainRegistrationState.REGISTERED;
+                            this.alfrescoRepository.updateAlfrescoNodeWith(entry.getId(), registrationState);
+                        } catch (Throwable throwable) {
```

Comment:
Throwable?

Reply:
All over the place unfortunately, changed it to Exception. A bit better, but I hope we can revise it in the next development phases.

Reply:
agree